File: app/src/test/java/com/vrem/wifianalyzer/wifi/model/SortByTest.java
/*
 *    Copyright (C) 2015 - 2016 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package com.vrem.wifianalyzer.wifi.model;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class SortByTest {

    @Test
    public void testSortByNumber() throws Exception {
        assertEquals(3, SortBy.values().length);
    }

    @Test
    public void testComparator() throws Exception {
        assertTrue(SortBy.STRENGTH.comparator() instanceof SortBy.StrengthComparator);
        assertTrue(SortBy.SSID.comparator() instanceof SortBy.SSIDComparator);
        assertTrue(SortBy.CHANNEL.comparator() instanceof SortBy.ChannelComparator);
    }

    @Test
    public void testFind() throws Exception {
        assertEquals(SortBy.STRENGTH, SortBy.find(-1));
        assertEquals(SortBy.STRENGTH, SortBy.find(3));

        assertEquals(SortBy.STRENGTH, SortBy.find(SortBy.STRENGTH.ordinal()));
        assertEquals(SortBy.SSID, SortBy.find(SortBy.SSID.ordinal()));
        assertEquals(SortBy.CHANNEL, SortBy.find(SortBy.CHANNEL.ordinal()));
    }
}

File: app/src/test/kotlin/com/vrem/wifianalyzer/wifi/model/SortByTest.kt
/*
 * WiFiAnalyzer
 * Copyright (C) 2015 - 2025 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>
 */
package com.vrem.wifianalyzer.wifi.model

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class SortByTest {
    @Test
    fun sortBy() {
        assertThat(SortBy.entries)
            .hasSize(3)
            .containsExactly(SortBy.STRENGTH, SortBy.SSID, SortBy.CHANNEL)
    }

    @Test
    fun sortByOrdinal() {
        assertThat(SortBy.STRENGTH.ordinal).isEqualTo(0)
        assertThat(SortBy.SSID.ordinal).isEqualTo(1)
        assertThat(SortBy.CHANNEL.ordinal).isEqualTo(2)
    }

    @Test
    fun comparator() {
        assertThat(
            SortBy.STRENGTH.sort.javaClass
                .isInstance(sortByStrength()),
        ).isTrue
        assertThat(
            SortBy.SSID.sort.javaClass
                .isInstance(sortBySSID()),
        ).isTrue
        assertThat(
            SortBy.CHANNEL.sort.javaClass
                .isInstance(sortByChannel()),
        ).isTrue
    }
}
